### ISSUE TITLE
Errors Opening Bags

### ISSUE THREAD

**Issue by dccjr3927** [Jaliborc/Scrap#173]
I am getting the following errors when opening my bag with Scrap enabled.
ERROR #1
Message: Interface/AddOns/Scrap/addons/main/spotlight.lua:54: attempt to call global 'GetContainerItemID' (a nil value)
Time: Thu Nov 17 16:55:43 2022
Count: 1
Stack: Interface/AddOns/Scrap/addons/main/spotlight.lua:54: attempt to call global 'GetContainerItemID' (a nil value)
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:54: in function `UpdateButton'
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:47: in function `call'
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:73: in function `IterateFrames'
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:46: in function `UpdateContainer'
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:32: in function <Interface/AddOns/Scrap/addons/main/spotlight.lua:32>
[string "=[C]"]: in function `Update'
[string "@Interface/FrameXML/ContainerFrame.lua"]:1133: in function `ContainerFrame_GenerateFrame'
[string "@Interface/FrameXML/ContainerFrame.lua"]:138: in function `ToggleBag_Individual'
[string "@Interface/FrameXML/ContainerFrame.lua"]:156: in function <Interface/FrameXML/ContainerFrame.lua:148>
[string "=[C]"]: in function `ToggleBag'
[string "@Interface/FrameXML/ContainerFrame.lua"]:107: in function `ToggleBackpack_Individual'
[string "@Interface/FrameXML/ContainerFrame.lua"]:119: in function <Interface/FrameXML/ContainerFrame.lua:111>
[string "=[C]"]: in function `ToggleBackpack'
[string "@Interface/FrameXML/ContainerFrame.lua"]:290: in function <Interface/FrameXML/ContainerFrame.lua:279>
[string "=[C]"]: in function `OpenBackpack'
[string "@Interface/FrameXML/ContainerFrame.lua"]:1826: in function <Interface/FrameXML/ContainerFrame.lua:1825>
[string "@Interface/FrameXML/ContainerFrame.lua"]:1888: in function `ToggleAllBags'
[string "OPENALLBAGS"]:1: in function <[string "OPENALLBAGS"]:1>

Locals: self = <table> {
 SendSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:82
 RegisterMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:85
 UnregisterMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:148
 IterateFrames = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:69
 SendMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:54
 UnregisterSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:78
 NewIcon = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:95
 UpdateAll = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:39
 UpdateButton = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:51
 RegisterEvent = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:85
 UnregisterAllMessages = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:169
 Icons = <table> {
 }
 Glows = <table> {
 }
 NewGlow = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:83
 OnEnable = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:24
 UpdateContainer = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:45
 RegisterSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:74
 NewModule = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:68
 UnregisterAllEvents = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:169
 UnregisterEvent = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:148
 Tag = "SCRAP_"
}
frame = ContainerFrame1 {
 0 = <userdata>
 CheckAuthenticatorSlotsHelpTip = <function> defined @Interface/FrameXML/ContainerFrame.lua:1062
 UpdateMoneyFrame = <function> defined @Interface/FrameXML/ContainerFrame.

ERROR#2
Message: Interface/AddOns/Scrap/addons/main/spotlight.lua:54: attempt to call global 'GetContainerItemID' (a nil value)
Time: Thu Nov 17 16:55:43 2022
Count: 4
Stack: Interface/AddOns/Scrap/addons/main/spotlight.lua:54: attempt to call global 'GetContainerItemID' (a nil value)
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:54: in function `UpdateButton'
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:47: in function `call'
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:73: in function `IterateFrames'
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:46: in function `UpdateContainer'
[string "@Interface/AddOns/Scrap/addons/main/spotlight.lua"]:32: in function <Interface/AddOns/Scrap/addons/main/spotlight.lua:32>
[string "=[C]"]: in function `Update'
[string "@Interface/FrameXML/ContainerFrame.lua"]:1133: in function `ContainerFrame_GenerateFrame'
[string "@Interface/FrameXML/ContainerFrame.lua"]:206: in function <Interface/FrameXML/ContainerFrame.lua:200>
[string "@Interface/FrameXML/ContainerFrame.lua"]:222: in function <Interface/FrameXML/ContainerFrame.lua:214>
[string "=[C]"]: in function `OpenBag'
[string "@Interface/FrameXML/ContainerFrame.lua"]:1832: in function <Interface/FrameXML/ContainerFrame.lua:1825>
[string "@Interface/FrameXML/ContainerFrame.lua"]:1888: in function `ToggleAllBags'
[string "OPENALLBAGS"]:1: in function <[string "OPENALLBAGS"]:1>

Locals: self = <table> {
 SendSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:82
 RegisterMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:85
 UnregisterMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:148
 IterateFrames = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:69
 SendMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:54
 UnregisterSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:78
 NewIcon = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:95
 UpdateAll = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:39
 UpdateButton = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:51
 RegisterEvent = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:85
 UnregisterAllMessages = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:169
 Icons = <table> {
 }
 Glows = <table> {
 }
 NewGlow = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:83
 OnEnable = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:24
 UpdateContainer = <function> defined @Interface/AddOns/Scrap/addons/main/spotlight.lua:45
 RegisterSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:74
 NewModule = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:68
 UnregisterAllEvents = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:169
 UnregisterEvent = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:148
 Tag = "SCRAP_"
}
frame = ContainerFrame5 {
 0 = <userdata>
 CheckAuthenticatorSlotsHelpTip = <function> defined @Interface/FrameXML/ContainerFrame.lua:1062
 UpdateMoneyFrame = <function> defined @Interface/FrameXML/ContainerFrame.lua:730
 UpdateFilterIcon = <function> defined @Interface/FrameXML/ContainerFrame.lua:922
 GetTitleText = <function> defined @Interface/SharedXML/PortraitFrame.lua:28
 GetPaddingWidth = <function> defined @Interface/FrameXML/ContainerFrame.lua:849
 SetPortraitTexCoord = <function> defined @Interface/SharedXML/PortraitFrame.lua:87
 size = 30
 CheckUpdateDynamicContents = <function> defi

ERROR#3 (The Count just keeps going up.)
Message: Interface/AddOns/Scrap/addons/main/main.lua:228: attempt to call global 'GetContainerNumSlots' (a nil value)
Time: Thu Nov 17 16:58:44 2022
Count: 7630
Stack: Interface/AddOns/Scrap/addons/main/main.lua:228: attempt to call global 'GetContainerNumSlots' (a nil value)
[string "@Interface/AddOns/Scrap/addons/main/main.lua"]:228: in function `GuessBagSlot'
[string "@Interface/AddOns/Scrap/addons/main/main.lua"]:216: in function `GuessLocation'
[string "@Interface/AddOns/Scrap/addons/main/main.lua"]:139: in function `IsFiltered'
[string "@Interface/AddOns/Scrap/addons/main/main.lua"]:86: in function `IsJunk'
[string "@Interface/AddOns/Sorted_Scrap/Sorted_Scrap.lua"]:65: in function `preSortingFunc'
[string "@Interface/AddOns/Sorted/List.lua"]:301: in function `SortTable'
[string "@Interface/AddOns/Sorted/List.lua"]:145: in function `UpdateDisplayedEntryData'
[string "@Interface/AddOns/Sorted/List_Items.lua"]:1053: in function <Interface/AddOns/Sorted/List_Items.lua:1044>
[string "@Interface/AddOns/Sorted/List_Items.lua"]:1087: in function <Interface/AddOns/Sorted/List_Items.lua:1086>

Locals: self = <table> {
 SendSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:82
 PrintMoney = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:275
 NewModule = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:68
 IterateJunk = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:98
 IsBetterEquip = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:204
 tip = ScrapTooltip {
 }
 charsets = <table> {
 }
 GuessLocation = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:215
 IsStandardQuality = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:175
 ScanLine = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:267
 Spotlight = <table> {
 }
 IsCombatItem = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:179
 Tag = "SCRAP_"
 IsLowLevel = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:171
 DestroyJunk = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:121
 ToggleJunk = <function> defined @Interface/AddOns/Sorted_Scrap/Sorted_Scrap.lua:43
 Print = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:279
 junk = <table> {
 }
 sets = <table> {
 }
 baseList = <table> {
 }
 GuessBagSlot = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:223
 SendMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:54
 OnSettings = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:66
 UnregisterAllEvents = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:169
 UnregisterSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:78
 IsJunk = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:84
 LoadTip = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:250
 RegisterMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:85
 RegisterSignal = <function> defined @Interface/AddOns/Scrap/libs/WildAddon-1.0/WildAddon-1.0.lua:74
 UnregisterAllMessages = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:169
 UnregisterMessage = <function> defined @Interface/AddOns/!KalielsTracker/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:148
 BelongsToSet = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:246
 IsOtherClass = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:237
 OnEnable = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:55
 Name = "Scrap"
 IsLowEquip = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:183
 IsGray = <function> defined @Interface/AddOns/Scrap/addons/main/main.lua:167
 IsFiltered = <function>

**Comment by aurel-burik** [comment]
Also having the same issue

**Comment by Tamales88** [comment]
Same

**Comment by Kirri777** [comment]
Fix for 10.0.0.2
Change these variables / functions:

/Scrap/addons/main/spotlight.lua
```
GetContainerItemID -> C_Container.GetContainerItemID
```

/Scrap/addons/main/main.lua
```
GetContainerItemID -> C_Container.GetContainerItemID
GetContainerNumSlots -> C_Container.GetContainerNumSlots
PickupContainerItem -> C_Container.PickupContainerItem
```

/Scrap/addons/merchant/learning.lua
```
UseContainerItem -> C_Container.UseContainerItem
GetContainerItemInfo -> C_Container.GetContainerItemInfo
GetContainerItemID -> C_Container.GetContainerItemID
GetContainerItemLink -> C_Container.GetContainerItemLink
hooksecurefunc('UseContainerItem', function(...) -> hooksecurefunc(C_Container, 'UseContainerItem', function(...)
```

/Scrap/addons/merchant/button.lua
```
GetContainerItemInfo -> C_Container.GetContainerItemInfo
UseContainerItem -> C_Container.UseContainerItem
PickupContainerItem -> C_Container.PickupContainerItem

function Button:GetReport()
	local qualities = {}
	local total = 0

	for bag, slot, id in Scrap:IterateJunk() do
		local containerInfo = C_Container.GetContainerItemInfo(bag, slot)
		local locked = containerInfo and containerInfo.isLocked;
		local quality = containerInfo and containerInfo.quality;
		local itemCount = containerInfo and containerInfo.stackCount;
		if not locked then
			qualities[quality] = (qualities[quality] or 0) + itemCount
			total = total + itemCount * (select(11, GetItemInfo(id)) or 0)
		end
	end


	return total, qualities
end
```
